### ISSUE TITLE
Re-enable metadata save when previewing a post

### ISSUE THREAD

**Issue by talldan** [WordPress/gutenberg#14900]
**Description**
#14877 removed the saving of metaboxes when the preview button is clicked. It did so due to an issue where changes to a published post would not be reflected in the preview when metaboxes are enabled.

This issue tracks reintroducing that functionality in a way where the preview still works.

**Comment by paaljoachim** [comment]
Is this issue still relevant? I see a PR has been merged.

**Comment by jonaskuske** [comment]
@paaljoachim 

Yep, still very relevant. The only merged PR I see is the one that *disabled* saving of metadata, so we still need a solution to re-enable it.

**Comment by paaljoachim** [comment]
Thank you @jonaskuske Jonas!

Dan @talldan is this something you or someone else could take a closer look at?
I'll ping Greg as well, as he might have some ideas.
@gziolo

**Comment by gziolo** [comment]
> I'll ping Greg as well, as he might have some ideas.

Unfortunately, I don't know much about metadata.

**Comment by adamsilverstein** [comment]
Some related discussion on how to get this working in this and follow up comments: https://github.com/WordPress/gutenberg/issues/16006#issuecomment-660261349 I haven't had a chance to work on this yet.

**Comment by korynorthrop** [comment]
It seems that metadata changes can be reflected when previewing a post if you utilize the `revisions_enabled` argument when registering a new meta key as mentioned in [this ticket](https://core.trac.wordpress.org/ticket/20299#comment:64) and in the [`Parameters` section](https://developer.wordpress.org/reference/functions/register_meta/#parameters) of the `register_meta` function.

**Comment by adamsilverstein** [comment]
> It seems that metadata changes can be reflected when previewing a post if you utilize the `revisions_enabled` argument when registering a new meta key as mentioned in [this ticket](https://core.trac.wordpress.org/ticket/20299#comment:64) and in the [`Parameters` section](https://developer.wordpress.org/reference/functions/register_meta/#parameters) of the `register_meta` function.

Right, this should work for any meta field - we would just need to enable meta revisions for meta when generating a preview autosave. Plugins can already opt into this. To enable automatically, we would also need to verify temporarily revisioned meta wasn't restored if the auto save was restored.

**Comment by adamsilverstein** [comment]
@korynorthrop Do you know if this is this still an issue with ACF out of the box or do they handle it automatically already? Are some other common plugins not working correctly?

**Comment by korynorthrop** [comment]
@adamsilverstein I can't speak to ACF of other common plugins; I can only confirm from my nonprofit's custom plugin that we maintain that has some metadata created via `register_post_meta` which did not show updates when previewing posts, but now do once I added the `revisions_enabled` argument.

**Comment by adamsilverstein** [comment]
Thanks for confirming. Maybe this is something we can improve docs for rather than trying to "fix" directly.
